python/python_helper_sympy.py

```diff
@@ -1,30 +1,70 @@
 import sympy
 import ast
-import sys
+from dataclasses import dataclass
+from itertools import chain
+from typing import Any, cast, Dict, List, Tuple, TypedDict, Literal, Optional
+
+SympyMapT = Dict[str, Any]
+ASTWhitelistT = Tuple[Any, ...]
+
+class SympyJson(TypedDict):
+    "A class with type signatures for the sympy json dict"
+    _type: Literal['sympy']
+    _value: str
+    _variables: List[str]
+
+class LocalsForEval(TypedDict):
+    "A class with type signatures for the locals_for_eval dict"
+
+    functions: SympyMapT
+    variables: SympyMapT
+    helpers: SympyMapT
+
```

Comment:
inconsistent spacing between these two class definitions.

Reply:
Actually, I don't know if we should be bothering to define these at all. They are only used once, and they are defined by a dict literal, so a type inferencer should be able to do all the type checking just fine without this extra clutter.

Reply:
Type inferences are always treated as generously as possible. Without this, the only type hint we can give is for a `Dict[str, Any]` (in fact, I think the inferred type may just be a `Dict`). This is actually why the typechecker will usually ask you for an explicit type hint on every dict that gets created.

Reply:
this is very annoying compared to TypeScript, where the the typechecker would just infer the type from the literal.

Reply:
Yes, all Python typecheckers are written to make interop with non-type-hinted code possible without changes. The downside of this is having to be very explicit with everything.